aten/src/ATen/native/native_functions.yaml

```diff
@@ -4877,11 +4877,21 @@
   dispatch:
     CPU, CUDA: scatter_reduce_
 
+- func: scatter.reduce(Tensor self, int dim, Tensor index, Tensor src, *, str reduce) -> Tensor
+  variants: function, method
+  dispatch:
+    CPU, CUDA: scatter_reduce
```

Comment:
Should there also be an out variant of this?

Reply:
I don't see why not! I will add them.

Reply:
If so, consider making it structured! https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md